ballerina/document_types.bal

```diff
@@ -0,0 +1,134 @@
+
+final string:RegExp FILE_URL = re `HTTP_URL|FTP_URL|FTPS_URL|FTP_WITH_AUTH_URL|GOOGLE_DRIVE_FILE_URL|GOOGLE_DRIVE_DOWNLOAD_URL|GOOGLE_DRIVE_FOLDER_URL|AMAZON_S3_URL|AMAZON_S3_CUSTOM_DOMAIN_URL|AMAZON_S3_PRESIGNED_URL|ONEDRIVE_FILE_URL|ONEDRIVE_DOWNLOAD_URL|GITHUB_RAW_URL|GITHUB_REPO_FILE_URL|DROPBOX_URL|DROPBOX_DOWNLOAD_URL|LOCAL_FILE_WINDOWS_URL|LOCAL_FILE_UNIX_URL`;
+
+# Enumeration of supported document types
+public enum DocumentKind {
+    # Text document type
+    TEXT = "text",
+    # Image document type  
+    IMAGE = "image",
+    # Audio document type
+    AUDIO = "audio",
+    # File document type
+    FILE = "file"
+}
+
+# Represents a URL pointing to the content of a document
+@constraint:String {
+    pattern: {
+        value: FILE_URL,
+        message: "A valid URL pointing to the document content, must be either http, ftp or google drive URL."
+    }
+}
+public type Url string;
+
+# Record type for file ID reference of a document
+public type FileId record {|
+    # Unique identifier for the file
+    string fileId;
+|};
+
+# Represents additional metadata associated with documents
+public type DocumentMetaData record {|
+    # MIME type specification for the file
+    string mimeType?;
+    # File name for the document
+    string fileName?;
+    # File size in bytes
+    decimal fileSize?;
+    # Creation timestamp of the file
+    time:Utc createdAt?;
+    # Modification timestamp of the file
+    time:Utc modifiedAt?;
+    json...;
+|};
+
+# Represents the common structure for all document types
+public type Document record {|
+    # The type of document (text, image, audio, or file)
+    string 'type;
+    # Metadata associated with the document
+    DocumentMetaData metadata?;
+    # The actual content of the document
+    anydata content;
+|};
+
+# Represents documents containing plain text content
+public type TextDocument record {|
+    *Document;
+    # Fixed type for the text document
+    readonly TEXT 'type = TEXT;
+    # The text content of the document
+    string content;
+|};
+
+# Represents documents containing image data
+public type ImageDocument record {|
+    *Document;
+    # Fixed type identifier for image documents
+    readonly IMAGE 'type = IMAGE;
+    # Image content - can be either a URL reference or binary data
+    Url|byte[] content;
+|};
+
+# Represents documents containing audio data
+public type AudioDocument record{|
+    *Document;
+    # Fixed type identifier for audio documents
+    readonly AUDIO 'type = AUDIO;
+    # Audio format specification (defaults to WAV)
+    string format = "wav";
+    # Audio content - can be either a URL reference or binary data
+    Url|byte[] content;
+|};
+
+# Represents generic file documents with various content sources
+public type FileDocument record {|
+    *Document;
+    # Fixed type identifier for file documents
+    readonly FILE 'type = FILE;
+    # File content - can be URL, binary data, or file ID reference
```

Comment:
Inconsistent description with URL - URL vs URL reference. I'm okay with URL, but has to be consistent.